### PR TITLE
Enable dependabot in Mimir

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by ying-jeanne** [comment]
This PR would add dependabot for go and docker dependencies in Mimir, dockerfile with no standard name is supported as well https://github.com/dependabot/dependabot-core/issues/4449

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 * [CHANGE] Compactor: the deprecated configuration parameter `-compactor.consistency-delay` has been removed. #5050
 * [CHANGE] Store-gateway: the deprecated configuration parameter `-blocks-storage.bucket-store.consistency-delay` has been removed. #5050
 * [CHANGE] The configuration parameter `-blocks-storage.bucket-store.bucket-index.enabled` has been deprecated and will be removed in Mimir 2.11. Mimir is running by default with the bucket index enabled since version 2.0, and starting from the version 2.11 it will not be possible to disable it. #5051
+* [CHANGE] Enable dependabot in Mimir. #5121
```
Comment (charleskorn) [inline CHANGELOG.md:22]: Given this isn't a user-visible change, I don't think we need a changelog entry.

**Review by charleskorn — APPROVED** [review]
Not blocking: it might be worth starting with just a few targets (eg. the main Mimir Dockerfile and our Go dependencies) and expanding the scope once we're happy with those rather than turning on everything to begin with.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    schedule:
+      interval: "weekly"
+      time: "06:04"
+    open-pull-requests-limit: 5
+  - package-ecosystem: docker
+    directory: /mimir-build-image
+    schedule:
+      interval: "weekly"
+      time: "06:06"
+    open-pull-requests-limit: 5
```
Comment (dimitarvdimitrov) [inline .github/dependabot.yml:31]: If we update the build image dockerfile shouldn't we also update the usages of the build image?

Comment (ying-jeanne) [reply .github/dependabot.yml:31]: for the moment to update the build image, we need to follow manual steps https://github.com/grafana/mimir/blob/8b8d822f39a4bb95189ef72cad9d2596f7225575/docs/internal/how-to-update-the-build-image.md?plain=1#L3, you are right, I think the best might be automate the build image update once some PR changed config there, build and push automatically. will remove this for the moment

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thanks for setting this up

**Pull request merged**